publisher-sdk/src/main/java/com/criteo/publisher/logging/RemoteHandler.kt

```diff
@@ -20,24 +20,28 @@ import com.criteo.publisher.SafeRunnable
 import com.criteo.publisher.csm.ConcurrentSendingQueue
 import com.criteo.publisher.logging.RemoteLogRecords.RemoteLogLevel.Companion.fromAndroidLogLevel
 import com.criteo.publisher.model.Config
+import com.criteo.publisher.privacy.ConsentData
 import java.util.concurrent.Executor
 
 internal class RemoteHandler(
     private val remoteLogRecordsFactory: RemoteLogRecordsFactory,
     private val sendingQueue: ConcurrentSendingQueue<RemoteLogRecords>,
     private val config: Config,
-    private val executor: Executor
+    private val executor: Executor,
+    private val consentData: ConsentData
 ) : LogHandler {
   override fun log(tag: String, logMessage: LogMessage) {
-    fromAndroidLogLevel(logMessage.level)?.takeIf { it >= config.remoteLogLevel } ?: return
+    if (consentData.isConsentGiven()) {
```

Comment:
Nit: there is already a lot of `}` at the end of the file. What do you think about writing the condition as a guard instead ?

```
if (!consentGiven) {
    return
}

// Do the job
```